presto-main/src/test/java/io/prestosql/sql/planner/TestPredicatePushdown.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.prestosql.metadata.Metadata;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.prestosql.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.anyTree;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.equiJoinClause;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.filter;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.join;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.project;
+import static io.prestosql.sql.planner.assertions.PlanMatchPattern.tableScan;
+import static io.prestosql.sql.planner.plan.JoinNode.Type.INNER;
+
+public class TestPredicatePushdown
+        extends AbstractPredicatePushdownTest
+{
+    private final Metadata metadata = createTestMetadataManager();
+
+    public TestPredicatePushdown()
+    {
+        super(ImmutableMap.of(ENABLE_DYNAMIC_FILTERING, "true"));
+    }
+
+    @Test
+    @Override
+    public void testCoercions()
+    {
+        // Ensure constant equality predicate is pushed to the other side of the join
+        // when type coercions are involved
+
+        // values have the same type (varchar(4)) in both tables
+        assertPlan(
+                "WITH " +
+                        "    t(k, v) AS (SELECT nationkey, CAST(name AS varchar(4)) FROM nation)," +
```

Comment:
`s/"    /"/`

Reply:
it's verbatim copy of this test method being moved

i will leave this as is